samza-core/src/main/java/org/apache/samza/clustermanager/ContainerManager.java

```diff
@@ -285,34 +300,40 @@ void handleExpiredRequest(String processorId, String preferredHost,
    * @param containerAllocator to request physical resources
    */
   public void registerContainerPlacementAction(ContainerPlacementRequestMessage requestMessage, ContainerAllocator containerAllocator) {
-    LOG.info("Received a ContainerPlacement action request: {}", requestMessage);
     String processorId = requestMessage.getProcessorId();
     String destinationHost = requestMessage.getDestinationHost();
-    Pair<ContainerPlacementMessage.StatusCode, String> actionStatus =
-        checkValidControlAction(processorId, destinationHost, requestMessage.getUuid());
-
+    // Is the action ready to be de-queued and taken or it needs to wait to be executed in future
+    if (!deQueueAction(requestMessage)) {
+      LOG.info("ContainerPlacement request is en-queued metadata: {}", requestMessage);
```

Comment:
Does this imply that a previous placement-action exists or container is not yet running?

Reply:
yes, added that message to logging